File: Filmes-FRONTEND/src/core/models/all/Diretor.ts
import { FilmeDiretor } from "./FilmeDiretor";
import { Genero } from "./Genero";

export class Diretor implements Shared.IEntity{
    id: number = 0;
    nome: string = '';
    dataNascimento: string = '';
    nacionalidade: string = '';
    generoId: number = 0;
    genero!: Genero;
    filmes: FilmeDiretor[]=[];

    constructor(model?: Diretor){

        if(!model)
            return;

        this.id = model.id;
        this.nome = model.nome;
        this.dataNascimento = model.dataNascimento;
        this.nacionalidade = model.nacionalidade;
        this.generoId = model.generoId;
        this.genero = model.genero;
        this.filmes = model.filmes
    }
}

File: Filmes-API/Filmes.DOMAIN/Entity/All/Diretor.cs
using Filmes.DOMAIN.Interface;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Filmes.DOMAIN.Entity.All
{
    [Table("Diretor", Schema = "All")]
    public class Diretor: IEntity
    {

        public long Id { get; set; }
        [MaxLength(45), Required] public string Nome { get; set; }
        public DateTime DataNascimento { get; set; }
        public string Nacionalidade { get; set; }
        public long GeneroId { get; set; }
        public Genero Genero { get; private set; }
        public ICollection<FilmeDiretor> Filmes { get; set; }
        public Diretor()
        {
            Filmes = new List<FilmeDiretor>();
        }
    }
}
